src/main/webapp/WEB-INF/static/js/trialmanager/datasetService.js

```diff
@@ -220,29 +220,43 @@
 		}]);
 
 
-	datasetsApiModule.factory('experimentDesignService', ['$http', '$q', 'studyContext', 'serviceUtilities',
-		function ($http, $q, studyContext, serviceUtilities) {
+	datasetsApiModule.factory('experimentDesignService', ['$http', '$q', 'studyContext', 'serviceUtilities', '$uibModal',
+		function ($http, $q, studyContext, serviceUtilities, $uibModal) {
 
 			var BASE_CROP_URL = '/bmsapi/crops/' + studyContext.cropName;
-			var BASE_STUDY_URL = BASE_CROP_URL  +  '/studies/';
+			var BASE_STUDY_URL = BASE_CROP_URL + '/studies/';
 
 			var experimentDesignService = {};
 			var successHandler = serviceUtilities.restSuccessHandler,
 				failureHandler = serviceUtilities.restFailureHandler;
 
+			experimentDesignService.openSelectEnvironmentToGenerateModal = function (experimentDesignInput) {
```

Comment:
@mderamos this opening of modal code can be moved to generateDesignCtrl, so that experimentDesignService only has logic for calling the APIs.